libraries/chain/db_block.cpp

```diff
@@ -746,5 +784,41 @@ bool database::before_last_checkpoint()const
 {
    return (_checkpoints.size() > 0) && (_checkpoints.rbegin()->first >= head_block_num());
 }
+   
+vector< custom_authority_object > database::get_custom_authorities_by_account( account_id_type account ) const
+{
+   const auto& authority_by_account = get_index_type<custom_authority_index>().indices().get<by_account>();
+   
+   vector<custom_authority_object> result;
+   
+   auto itr = authority_by_account.find(account);
+   while(itr != authority_by_account.end() && (*itr).account == account)
```

Comment:
Probably inefficient once many items in the index.
Erm, also, shouldn't the list be indexed `by_account`, already?

Reply:
Hm, may be, I'll add test for it.

Reply:
@pmconrad what is your take on the possible perfomance issue?

Reply:
@sschiessl-bcp I've commented above as well as somewhere else in this PR about iteration. The internal interface / function need to be redesigned IMHO, not only this line.

Reply:
I've fixed that. Replaced iteration with equal_range function call.